packages/amplify-provider-awscloudformation/src/iterative-deployment/stack-progress-printer.ts

```diff
@@ -59,21 +59,29 @@ export class StackProgressPrinter implements IStackProgressPrinter {
       const category = this.eventMap.eventToCategories.get(event.LogicalResourceId);
       if (category && !this.categoriesPrinted.includes(category)) {
         const categoryItem = this.eventMap.categories.find(it => it.name === category);
-        progressBarsConfigs.push({
-          name: categoryItem.name,
-          value: 0,
-          total: categoryItem.size,
-          payload: {
-            progressName: categoryItem.name,
-            envName: this.eventMap.envName,
-          },
-        });
-        this.categoriesPrinted.push(category);
+        if (categoryItem) {
+          progressBarsConfigs.push({
+            name: categoryItem.name,
+            value: 0,
+            total: categoryItem.size,
+            payload: {
+              progressName: categoryItem?.name,
+              envName: this.eventMap.envName,
+            },
+          });
+          this.categoriesPrinted.push(category);
+        }
       }
       this.progressBars.create(progressBarsConfigs);
     }
   };
 
+  updateIndexInHeader = (currentIndex: number, totalIndices: number): void => {
+    this.progressBars.updatePrefixText(
+      `Deploying iterative update ${currentIndex} of ${totalIndices} into ${this.eventMap.envName} environment. This will take a few minutes.`,
+    );
+  }
```

Comment:
Should this be `updateRelativeProgressCounter` or something like this?
Or perhaps should be this be `onProgress(completed, total)` ? The fact that this updates header is implementation details of this component.

Reply:
Great catch, the class shouldn't leak implementation details. Added to backlog refactor task.